src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/ComponentViewLocationExpanderProvider.cs

```diff
@@ -112,6 +136,27 @@ public void PopulateValues(ViewLocationExpanderContext context)
                 }
 
                 result.AddRange(moduleComponentViewLocations);
+
+                if (!_memoryCache.TryGetValue(CacheKey, out string[] moduleWithPagesComponentViewLocations))
+                {
+                    var enabledIds = _extensionManager.GetFeatures().Where(f => _shellDescriptor
```

Comment:
So here i would have used only one cache entry so that we can re-use the same `enabledIds` and `enabledExtensionIds` to still build 2 local location collections by using `_modulesWithComponentViews` and `_modulesWithPagesComponentViews`, but then concat them to update an unique cache entry.

Reply:
> but then concat them to update an unique cache entry.

I didn't get this, l will update the PR, feel free to add your commits then after if not meet your suggestions